src/app/shared/change-password-dialog/password.service.ts

```diff
@@ -0,0 +1,50 @@
+/** Angular Imports */
+import { Injectable } from '@angular/core';
+import { ValidatorFn, AbstractControl, ValidationErrors } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+
+/**
+ * Password Check service.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class PasswordService {
+
+  /**
+   * @param {HttpClient} http Http Client to send requests.
+   */
+  constructor(private http: HttpClient) { }
+
+  /**
+   * Confirm Change Password of Users
+   * @param controlNameToCompare Form Control Name to be compared.
+   */
+  confirmPassword(controlNameToCompare: string): ValidatorFn {
+    return (c: AbstractControl): ValidationErrors|null => {
+        if (c.value == null || c.value.length === 0) {
+            return null;
+        }
+        const controlToCompare = c.root.get(controlNameToCompare);
+        if (controlToCompare) {
+            const subscription: Subscription = controlToCompare.valueChanges.subscribe(() => {
+                c.updateValueAndValidity();
+                subscription.unsubscribe();
+            });
+        }
+        return controlToCompare && controlToCompare.value !== c.value ? {'notequal': true} : null;
+    };
+  }
```

Comment:
The function can be executed in the `change-password-dialog.component.ts`, a separate file is not needed. Can you change this?

Reply:
This is being used in self service user password as well

Reply:
Muskan please dounle check validatior function thats the main thing

Reply:
Yuh. May I delete the service file ? But this will be needed in #670 too

Reply:
put this in dialog we'll use the dialog in 670 too